python/smashbox/utilities/__init__.py

```diff
@@ -626,23 +627,61 @@ def reset_server_log_file(force = False):
     cmd = '%s rm -rf %s/owncloud.log' % (config.oc_server_shell_cmd, config.oc_server_datadirectory)
     runcmd(cmd)
 
-def get_log_file():
-    """ Copies over the server log file
-
+def reset_diagnostics(force = False):
+    """ Deletes the existing server log file so that there is a clean
+        log file for the test run, and set neccesairly flags on the server for diagnostic log generation
     """
 
-    # download server log
+    if not force:
+        try:
+            if not config.oc_check_diagnostic_log:
+                return
+        except AttributeError: # allow this option not to be defined at all
+            return
+
+    logger.info('Initializing diagnostic log file')
     log_url = 'http'
     if config.oc_ssl_enabled:
         log_url += 's'
     log_url += '://' + config.oc_admin_user + ':' + config.oc_admin_password + '@' + config.oc_server
-    log_url += '/' + os.path.join(config.oc_root, 'index.php/settings/admin/log/download')
 
-    res = requests.get(log_url)
+    clean_log_url = log_url + '/' + os.path.join(config.oc_root, 'index.php/apps/diagnostics/log/clean')
+    res = requests.post(clean_log_url)
 
-    fatal_check(res.status_code == 200, 'Could not download the log file from the server, status code %i' % res.status_code)
+    fatal_check(res.status_code == 200, 'Could not clean the diagnostic log file from the server, status code %i' % res.status_code)
+    fatal_check(res.text == "null", 'Diagnostic app seems disabled, returned body %s' % res.text)
+
+def parse_log_file_lines(res):
+    data = []
+    if res is not None:
+        import json
+        for line in res.iter_lines():
+            data.append(json.loads(line))
```

Comment:
I am just looping over file lines and load JSON

![selection_178](https://cloud.githubusercontent.com/assets/13368647/24955060/32f7023a-1f82-11e7-8b6e-30931c632fba.jpg)